v-next/hardhat/src/internal/utils/package.ts

```diff
@@ -17,3 +18,27 @@ export async function getHardhatVersion(): Promise<string> {
 
   return packageJson.version;
 }
+
+export async function getLatestHardhatVersion(): Promise<string> {
+  if (cachedLatestHardhatVersion !== undefined) {
+    return cachedLatestHardhatVersion;
+  }
+
+  const packageJson: PackageJson = await readClosestPackageJson(
+    import.meta.url,
+  );
+
+  const packageName = packageJson.name;
+
+  const latestHardhat = await fetch(
```

Comment:
I think I misunderstood this other PR: https://github.com/NomicFoundation/hardhat/pull/5789

We just want to avoid using `fetch` everywhere, not during tests. Instead, we should be using `hardhat-utils/request`.

That's why I thought that a linter rule would be better.

Reply:
Yes! That makes so much more sense now. I replaced `fetch` with `getRequest` now and reverted the `doNotUseFetch` PR in https://github.com/NomicFoundation/hardhat/pull/5801

I also restored using network (i.e. accessing npm registry) during tests for now - I think we can explore it as a generic follow-up item.